src/osdproc.c

```diff
@@ -880,36 +986,48 @@ void set_home_altitude_if_unset() {
 
 // Set home distance and bearing, if we know where home is
 void set_home_distance_and_bearing() {
-  if (osd_got_home == 1) {
-    osd_home_distance = get_distance_from_home_in_meters();
-    osd_home_bearing = get_bearing_to_home_in_degrees();
+  if (get_osd_got_home() == 1) {
+    set_osd_home_distance(get_distance_from_home_in_meters());
+    set_osd_home_bearing(get_bearing_to_home_in_degrees());
   }
 }
 
 // direction - scale mode
 void draw_osd_linear_compass() {
-  if (!enabledAndShownOnPanel(eeprom_buffer.params.CWH_Tmode_en,
+  if (!enabledAndShownOnPanel(eeprom_buffer.params.CWH_Tmode_en, 
                               eeprom_buffer.params.CWH_Tmode_panel)) {
       return;
   }
+  
+  draw_linear_compass(osdproc_osd_state.osd_heading, 0, 120, 180, GRAPHICS_X_MIDDLE, eeprom_buffer.params.CWH_Tmode_posY, 15, 30, 5, 8, 0);
+}
+
+float get_new_updated_average_climb_rate() {
```

Comment:
the there a `get_old_average_climb_rate` or similar?
Otherwise I think we can call it `get_average_climb_rate`